custom/logistics/api.py

```diff
@@ -250,30 +261,38 @@ def sms_user_sync(self, ilsgateway_smsuser, username_part=None, password=None,
                 user.save()
         return user
 
-    def add_language_to_user(self, logistics_sms_user, domains=None):
-        if not domains:
-            domains = []
+    def _reassign_number(self, user, phone_number):
+        v = VerifiedNumber.by_phone(phone_number, include_pending=True)
+        if v.domain in self._get_logistics_domains():
+            v.delete()
+            user.save_verified_number(self.domain, phone_number, True)
```

Comment:
@kkrampa Do these tasks happen to run simultaneously for multiple users?  If so, it could be a race condition where 2 reassigns happen at the same time.  I think what's here is good, we would just have to wrap the entire call to _reassign_number in a CriticalSection:

    from dimagi.utils.couch import CriticalSection
    def do_reassign(...):
        with CriticalSection(['ils-reassign-phone-number-%s' % phone_number]):
            _reassign_number(...)